docker-compose.yml

```diff
@@ -0,0 +1,210 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+      target: frontend
+    container_name: oppia-angular-build
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+    environment:
+      - NODE_OPTIONS=--max_old_space_size=4096
+    healthcheck:
+      test: test "$(ls -A /app/oppia/dist)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 10m
+    command: >
+      sh -c '
+        trap "rm /app/oppia/node_modules/@angular/compiler-cli/ngcc/__ngcc_lock_file__" EXIT
+        /app/oppia/node_modules/.bin/ng build --watch
+      '
+
+  webpack-compiler:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+      target: frontend
+    container_name: oppia-webpack-compiler
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+    healthcheck:
+      test: test "$(ls -A /app/oppia/webpack_bundles)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 10m
+    entrypoint: node
+    command:
+      - "/app/oppia/node_modules/webpack/bin/webpack.js"
+      - "--config"
+      - "/app/oppia/webpack.dev.config.ts"
+      - "--watch"
+      - "--color"
+      - "--progress"
+
+  google-app-engine:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+      target: backend
+    container_name: oppia-app-engine
+    ports:
+      - "8181:8181"
+      - "8000:8000"
+    volumes:
+      - .:/app/oppia
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_third_party:/app/oppia/third_party
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
```

Comment:
since these dirs are generated within the dockerfile, we need to mount them in a vol so that the service can use it.
(tested on removing the FE proto files, so GAE is unable to import the required generated files)